File: Solutions/1669- Merge In Between Linked Lists/Solution.py
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        pointer1 = pointer2 = list1

        for _ in range(0,a-1):
            pointer1 = pointer1.next
        
        for _ in range(0,b):
            pointer2 = pointer2.next
        
        pointer1.next = list2
        while pointer1.next:
            pointer1 = pointer1.next
        
        pointer1.next = pointer2.next
        return list1

File: Solutions/1669- Merge In Between Linked Lists/Solution.java
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        var pointer1 = list1;
        var pointer2 = list1;

        while (a > 1) {
            pointer1 = pointer1.next;
            a--;
        }

        while (b > 0) {
            pointer2 = pointer2.next;
            b--;
        }

        pointer1.next = list2;
        while (pointer1.next != null) {
            pointer1 = pointer1.next;
        }

        pointer1.next = pointer2.next;
        return list1;

    }
}